### PR TITLE
Change default directory of the config file to be identical with the binary

### PR DESCRIPTION
**Pull request by ruflin**
See https://github.com/elastic/beats/pull/435 for more details

### CODE REVIEW

File: libbeat/cfgfile/cfgfile.go [hunk 1]
```diff
@@ -29,12 +29,13 @@ func ChangeDefaultCfgfileFlag(beatName string) error {
...
-		cliflag.DefValue = fmt.Sprintf(`C:\Program Files\%s\%s.yml`,
-			strings.Title(beatName), beatName)
-	} else {
-		cliflag.DefValue = fmt.Sprintf("/etc/%s/%s.yml", beatName, beatName)
+	path, err := filepath.Abs(filepath.Dir(os.Args[0]))
+	if err != nil {
+		return fmt.Errorf("Failed to set default config file location because the absolute path to %s could not be obtained. %v", os.Args[0], err)
 	}
+
+	cliflag.DefValue = fmt.Sprintf("%s/%s.yml", path, beatName)
```
Comment (andrewkroh) [inline libbeat/cfgfile/cfgfile.go]: Consider using `filepath.Join(path, beatName + .yml)` to get OS specific slashses here.

Comment (ruflin) [inline libbeat/cfgfile/cfgfile.go]: Adjusted

File: libbeat/cfgfile/cfgfile.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ var testConfig *bool
 func init() {
 	// The default config cannot include the beat name as it is not initialised when this
 	// function is called, but see ChangeDefaultCfgfileFlag
-	configfile = flag.String("c", "/etc/beat/beat.yml", "Configuration file")
+	configfile = flag.String("c", "beat.yml", "Configuration file")
```
Comment (ruflin) [inline libbeat/cfgfile/cfgfile.go]: That is a tricky one and we haven't found a solution yet. I don't think we should invest more time into this at the moment.

Comment (andrewkroh) [inline libbeat/cfgfile/cfgfile.go]: I thought the flag output was correct since we call `ChangeDefaultCfgfileFlag` before `flag.Parse()` in beat.go.

**Pull request merged**